File: backend/buildsmart/src/main/java/com/example/buildsmart/model/Staff.java
package com.example.buildsmart.model;

import java.time.LocalDateTime;
import java.util.Objects;

public class Staff {
    private String staffId;
    private String firstName;
    private String lastName;
    private String email;
    private String phoneNumber;
    private StaffRole role;
    private StaffStatus status;
    private LocalDateTime createdAt;
    private LocalDateTime lastLogin;
    private String createdBy;

    //Default constructor
    public Staff() {
        this.staffId = generateUniqueId();
        this.createdAt = LocalDateTime.now();
        this.status = StaffStatus.ACTIVE;
    }

    //Parameterized constructor
    public Staff(String firstName, String lastName, String email, String phoneNumber, StaffRole role, String createdBy){
        this();
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.role = role;
        this.createdBy = createdBy;
    }

    //Getters and Setters
    public String getStaffId(){
        return staffId;
    }

    public void setStaffId(String staffId){
        this.staffId = staffId;
    }

    public String getFirstName(){
        return firstName;
    }

    public void setFirstName(String firstName){
        this.firstName = firstName;
    }

    public String getLastName(){
        return lastName;
    }

    public void setLastName(String lastName){
        this.lastName = lastName;
    }

    public String getEmail(){
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPhoneNumber(){
        return phoneNumber;
    }

    public void setPhoneNumber(String phoneNumber){
        this.phoneNumber = phoneNumber;
    }

    public StaffRole getRole(){
        return role;
    }

    public void setRole(StaffRole role){
        this.role = role;
    }

    public StaffStatus getStatus() {
        return status;
    }

    public void setStatus(StaffStatus status){
        this.status = status;
    }

    public LocalDateTime getCreatedAt(){
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getLastLogin() {
        return lastLogin;
    }

    public void setLastLogin(LocalDateTime lastLogin) {
        this.lastLogin = lastLogin;
    }

    public String getCreatedBy() {
        return createdBy;
    }

    public void setCreatedBy(String createdBy) {
        this.createdBy = createdBy;
    }

    //Business methods
    public String getFullName(){
        return firstName + " " + lastName;
    }

    public boolean isActive(){
        return status == StaffStatus.ACTIVE;
    }

    public String generateUniqueId(){
        return "STAFF" + System.currentTimeMillis();
    }

    @Override
    public boolean equals(Object obj){
        if(this == obj)
            return true;
        if(obj == null || getClass() != obj.getClass())
            return false;
        Staff staff = (Staff) obj;
        return Objects.equals(staffId, staff.staffId);
    }

    @Override
    public int hashCode(){
        return Objects.hash(staffId);
    }

    @Override
    public String toString(){
        return "Staff{" +
                "staffId='" + staffId + '\'' +
                ", fullName='" + getFullName() + '\'' +
                ", email='" + email + '\'' +
                ", role=" + role +
                ", status=" + status +
                ", createdAt=" + createdAt +
                '}';
    }
}

File: frontend/staff-management-frontend/src/utils/Staff.js
// Staff roles mapping (matching Java enum)
export const StaffRole = {
    ADMIN: { displayName: "Admin", description: "Full system access and user management" },
    SITE_ENGINEER: { displayName: "Site Engineer", description: "Project and site management responsibilities" },
    DOCUMENT_CONTROL_MANAGER: { displayName: "Document Control Manager", description: "Document oversight and control" },
    SITE_STAFF: { displayName: "Site Staff", description: "Basic site operations and daily tasks" },
    BUDGET_PLANNING_TEAM: { displayName: "Budget Planning Team", description: "Budget analysis and planning" }
};

// Staff status mapping (matching Java enum)
export const StaffStatus = {
    ACTIVE: { displayName: "Active", description: "Staff member is currently active and can access the system" },
    INACTIVE: { displayName: "Inactive", description: "Staff member is temporarily inactive" },
    SUSPENDED: { displayName: "Suspended", description: "Staff member has been suspended from system access" },
    PENDING_ACTIVATION: { displayName: "Pending Activation", description: "New staff member awaiting activation" }
};

/**
 * Staff class matching Java backend Staff.java
 */
class Staff {
    constructor(firstName = '', lastName = '', email = '', phoneNumber = '', role = '', createdBy = '') {
        this.staffId = this.generateUniqueId();
        this.firstName = firstName;
        this.lastName = lastName;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.role = role;
        this.status = 'ACTIVE';
        this.createdAt = new Date();
        this.lastLogin = null;
        this.createdBy = createdBy;
    }

    generateUniqueId() {
        return 'STAFF' + Date.now();
    }

    getFullName() {
        return `${this.firstName} ${this.lastName}`;
    }

    isActive() {
        return this.status === 'ACTIVE';
    }

    toString() {
        return `Staff{staffId='${this.staffId}', fullName='${this.getFullName()}', email='${this.email}', role=${this.role}, status=${this.status}, createdAt=${this.createdAt.toISOString()}}`;
    }

    // Java-style getters and setters
    getStaffId() { return this.staffId; }
    setStaffId(staffId) { this.staffId = staffId; }

    getFirstName() { return this.firstName; }
    setFirstName(firstName) { this.firstName = firstName; }

    getLastName() { return this.lastName; }
    setLastName(lastName) { this.lastName = lastName; }

    getEmail() { return this.email; }
    setEmail(email) { this.email = email; }

    getPhoneNumber() { return this.phoneNumber; }
    setPhoneNumber(phoneNumber) { this.phoneNumber = phoneNumber; }

    getRole() { return this.role; }
    setRole(role) { this.role = role; }

    getStatus() { return this.status; }
    setStatus(status) { this.status = status; }

    getCreatedAt() { return this.createdAt; }
    setCreatedAt(createdAt) { this.createdAt = createdAt; }

    getCreatedBy() { return this.createdBy; }
    setCreatedBy(createdBy) { this.createdBy = createdBy; }
}

export default Staff;